modules/templates/proxy_request.rb

```diff
@@ -0,0 +1,55 @@
+require "proxy/util"
+require 'proxy/request'
+require 'uri'
+
+module Proxy::Templates
+  class ProxyRequest < ::Proxy::HttpRequest::ForemanRequest
+    include Proxy::Log
+
+    def call_template(method, path, env, params, body = '')
+      template_url = Proxy::Templates::Plugin.settings.template_url
+      proxy_ip = URI.parse(template_url).host
+      opts = params.clone.merge(:url => template_url)
+      opts.delete("path")
```

Comment:
Would it make sense to have a `BLACKLIST_PARAMETERS = ['template', 'kind', 'hostgroup', 'splat', 'captures']` and then iterate over it inside the code?